libs/datasets/manual_filter.py

```diff
@@ -10,88 +14,117 @@
-    drop_start_date = pd.to_datetime(config["start_date"])
+class Action(enum.Enum):
+    DROP_OBSERVATIONS = enum.auto()
+    ANNOTATE = enum.auto()
+
+
+class Filter(pydantic.BaseModel):
+    regions_included: List[RegionMaskOrRegion]
+    regions_excluded: Optional[List[RegionMaskOrRegion]] = None
+    fields_included: List[CommonFields]
+    action: Action
+    start_date: Optional[datetime.date] = None
+    internal_note: str
+    public_note: str
 
-    ts_in = dataset.timeseries_bucketed_wide_dates
+    @property
+    def tag(self) -> taglib.TagInTimeseries:
+        return taglib.KnownIssue(date=self.start_date, disclaimer=self.public_note)
+
+
+class Config(pydantic.BaseModel):
+    filters: List[Filter]
+
+
+CONFIG = Config(
+    filters=[
+        Filter(
+            regions_included=[RegionMask(AggregationLevel.COUNTY, states=["OK"])],
+            regions_excluded=[Region.from_fips("40109"), Region.from_fips("40143")],
+            fields_included=[CommonFields.CASES, CommonFields.DEATHS],
+            action=Action.DROP_OBSERVATIONS,
+            start_date="2021-03-15",
+            internal_note="https://trello.com/c/HdAKfp49/1139",
+            public_note="Something broke with the OK county data.",
+        )
+    ]
+)
 
-    if "field_group" in config:
-        fields = common_fields.FIELD_GROUP_TO_LIST_FIELDS[config["field_group"]]
-    else:
-        fields = config["fields"]
+
+def _partition_by_fields(
+    ts_in: pd.DataFrame, fields: List[CommonFields]
+) -> Tuple[pd.DataFrame, pd.DataFrame]:
+    """Partitions a DataFrame with timeseries wide dates into two: fields and others."""
+    timeseries.check_timeseries_wide_dates_structure(ts_in)
     mask_selected_fields = ts_in.index.get_level_values(PdFields.VARIABLE).isin(fields)
     ts_selected_fields = ts_in.loc[mask_selected_fields]
     ts_not_selected_fields = ts_in.loc[~mask_selected_fields]
+    return ts_selected_fields, ts_not_selected_fields
+
 
-    obsv_selected = ts_selected_fields.loc[:, ts_selected_fields.columns >= drop_start_date]
+def _filter_by_date(
+    ts_in: pd.DataFrame, *, drop_start_date: datetime.date
+) -> Tuple[pd.DataFrame, pd.DataFrame]:
+    """Removes observations from specified dates, returning the modified timeseries and
+    unmodified timeseries in separate DataFrame objects."""
+    timeseries.check_timeseries_wide_dates_structure(ts_in)
+    start_date = pd.to_datetime(drop_start_date)
+    obsv_selected = ts_in.loc[:, ts_in.columns >= start_date]
     mask_has_real_value_to_drop = obsv_selected.notna().any(1)
-    ts_to_filter = ts_selected_fields.loc[mask_has_real_value_to_drop]
-    ts_no_real_values_to_drop = ts_selected_fields.loc[~mask_has_real_value_to_drop]
+    ts_to_filter = ts_in.loc[mask_has_real_value_to_drop]
+    ts_no_real_values_to_drop = ts_in.loc[~mask_has_real_value_to_drop]
+    ts_filtered = ts_to_filter.loc[:, ts_to_filter.columns < start_date]
+    return ts_filtered, ts_no_real_values_to_drop
```

Comment:
Minor but it seems like rather than partitioning the data and returning both we could just filter the data and return the result and `mask_has_real_value_to_drop`, and both this code and the calling code might be simpler.  Feel free to leave it though.

Reply:
Okay, feel free to ignore.  I was basically reacting to the fact that we have code that goes out of its way to split the data apart and then the calling code just immediately puts them back together.  So returning a tuple of essentially {filtered result, affected rows} seemed simpler, but I may be missing something and don't care strongly.  🚢  it!